messaging/WFLY-14133_configurable_bridge_call_timeout.adoc

```diff
@@ -0,0 +1,136 @@
+= Add call-timeout attribute to the Artemis core bridge resource
+:author:            Tomas Hofman
+:email:             thofman@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:          messaging,jms
+:idseparator:       -
+
+== Overview
+
+The Artemis broker defines a call-timeout attribute on the core bridge, which sets a time out for blocking calls to the
+server. Unless modified, the call-timeout defaults to 30 seconds (this is the default value defined in Artemis).
+It is possible to programmatically modify this timeout by a lower level Artemis API, but it is not currently possible
+to modify with the high level API that Wildfly currently uses to create core bridges. It is currently not possible at all to
+modify this attribute by an EAP administrator.
+
+It was requested by GSS to make this attribute configurable via the Wildfly/JBoss EAP management model.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/EAP7-1622
+* https://issues.redhat.com/browse/WFLY-14133
+
+=== Related Issues
+
+* https://issues.apache.org/jira/browse/ARTEMIS-3074
+* https://issues.redhat.com/browse/JBEAP-20326
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mkopecky@redhat.com[Marek Kopecky]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Wildfly
+* ActiveMQ Artemis
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Make the `BridgeConfiguration#callTimeout` attribute configurable via the Wildfly management model.
+
+=== Nice-to-Have Requirements
+
+None.
+
+=== Non-Requirements
+
+None.
+
+== Implementation Plan
+
+=== Relevant Entities
+
+* `ActiveMQServerControl` is a part of the high level Artemis API used to control the embedded Artemis broker.
+* `ActiveMQServer` is part of a lower level Artemis API used to control the embedded Artemis broker.
+* `BridgeDefinition` is a Wildfly management model resource definition class describing the Artemis core bridge.
+* `BridgeAdd` is a Wildfly controller handler used to create the Artemis core bridges.
+
+=== Code Changes
+
+1. Add the `call-timeout` attribute to the `BridgeDefinition` class in the messaging-activemq subsystem.
+This also requires creating new schema version of the messaging management model (version 13).
++
+Attribute is of type long and its' default value is 30000 ms. The default value is defined in the Artemis code base
```

Comment:
so I guess it has a unit in milliseconds :)

Reply:
Yes, updated.